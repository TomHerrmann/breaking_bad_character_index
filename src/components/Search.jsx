import React, { useState, useEffect } from 'react';
import store from './store';

const Search = ({ onSearch, searchQuery }) => {
  return (
    <div className="search-container">
      <form onSubmit={(event) => event.preventDefault()}>
        <label>
          <input
            type="text"
            placeholder="Character Name"
            onChange={(event) => {
              onSearch(event);
            }}
            value={searchQuery}
          />
        </label>
      </form>
    </div>
  );
};

export default Search;
