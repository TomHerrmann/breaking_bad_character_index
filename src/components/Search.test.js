import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Search from './Search';

describe('<Search />', () => {
  let defaultProps;
  beforeEach(() => {
    defaultProps = {
      onSearch: jest.fn(),
      searchQuery: 'test',
    };
  });

  it('should render correctly', () => {
    const component = shallow(<Search {...defaultProps} />);
    expect(toJson(component)).toMatchSnapshot();
  });
  it('should call onSearch function onChange', () => {
    const component = shallow(<Search {...defaultProps} />);
    component
      .find('input')
      .props()
      .onChange();
    expect(defaultProps.onSearch).toHaveBeenCalled();
  });
  it('should call preventDefault function onSubmit', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const component = shallow(<Search {...defaultProps} />);
    component
      .find('form')
      .props()
      .onSubmit(event);
    expect(event.preventDefault).toHaveBeenCalled();
  });
});
