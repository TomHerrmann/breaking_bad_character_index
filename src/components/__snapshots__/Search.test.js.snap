// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should render correctly 1`] = `
<div
  className="search-container"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      <input
        onChange={[Function]}
        placeholder="Character Name"
        type="text"
        value="test"
      />
    </label>
  </form>
</div>
`;
